src/pages/index.js

```diff
@@ -0,0 +1,275 @@
+const profileNameInput = editModal.querySelector("#profile-name");
+const profileDescriptionInput = editModal.querySelector("#profile-description");
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+
+/*   CARD SELECTORS     */
+const addCardButton = document.querySelector(".profile__post-button");
+const addCardModal = document.querySelector("#post-modal");
+const cardForm = document.forms["new-post-form"];
+const cardLinkInput = addCardModal.querySelector("#card-link");
+const cardNameInput = addCardModal.querySelector("#card-name");
+const cardSubmitButton = addCardModal.querySelector(".modal__submit-button");
+const cardsList = document.querySelector(".cards__list");
+
+/*   MODAL & TEMPLATE SELECTORS     */
+const popups = document.querySelectorAll(".modal");
+const cardTemplate = document.querySelector("#card-template");
+
+/*   IMAGE MODAL SELECTORS   */
+const modalPreview = document.querySelector("#modal-preview");
+const modalImageName = modalPreview.querySelector(".modal__image-name");
+const modalImage = modalPreview.querySelector(".modal__image");
+
+/*    DELETE MODAL SELECTORS    */
+const modalDelete = document.querySelector("#modal-delete");
+const deleteForm = modalDelete.querySelector(".modal__form");
+const deleteSubmitButton = modalDelete.querySelector(".modal__submit-button");
+const cancelModalButton = document.querySelector(".modal__cancel-button");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "cb408f20-cef0-4ee4-a289-c2194b9dd6d5",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    avatarImage.src = userInfo.avatar;
+    cards.forEach((item) => {
+      renderCard(item, "append");
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleProfileFormSubmit(evt) {
+  // create a request function that returns a promise
+  function editUserInfo() {
+    // `return` lets us use a promise chain `then, catch, finally`
+    return api
+      .editUserInfo({
+        name: profileNameInput.value,
+        about: profileDescriptionInput.value,
+      })
+      .then((data) => {
+        profileName.textContent = data.name;
+        profileDescription.textContent = data.about;
+        closeModal(editModal);
+      });
+  }
+  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
+  handleSubmit(editUserInfo, evt);
+}
+
+function handleAvatarFormSubmit(evt) {
+  function editAvatar() {
+    return api
+      .editAvatar({
+        avatar: avatarLinkInput.value,
+      })
+      .then((data) => {
+        avatarImage.src = data.avatar;
+        disableButton(deleteSubmitButton, settings);
```

Comment:
`deleteSubmitButton` for the delete modal, but you should disable the `avatar` submit button in `.then`


```js

        disableButton(evt.submitter, settings);

```

`evt.submitter` is the submit button inside any submit handler